exec/rpc/src/main/java/org/apache/drill/exec/rpc/InvalidConnectionInfoException.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.rpc;
+
+/**
+ * Exception class to differentiate errors due to malformed connection string from client
+ */
+public class InvalidConnectionInfoException extends RpcException {
+  static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(InvalidConnectionInfoException.class);
+
+  private final String message;
+
+  public InvalidConnectionInfoException(String message) {
```

Comment:
Just use the normal facilities:

    public Invalid...( String message ) { super( message ); }
    ...
    e.getMessage( ) ...